examples/dht_node.rs

```diff
@@ -128,28 +126,16 @@ fn main() {
 
     let server_obj_c = server_obj.clone();
     let ping_sender = ping_wakeups.for_each(move |()| {
-            // println!("ping wakeup");
-            let kbucket_c = server_obj_c.borrow().kbucket.clone().iter();
-            let server_obj_c_c = server_obj_c.clone();
-
-            type LoopFuture = Box<Future<Item = future::Loop<KbucketIter, KbucketIter>, Error = ()>>;
-
-            let packet_sender = future::loop_fn(kbucket_c, move |mut iter| -> LoopFuture {
-                let kbucket_c_c = server_obj_c_c.borrow().kbucket.clone().iter();;
-                let peer = iter.next();
-                match peer {
-                    None => Box::new(Ok(future::Loop::Break(iter)).into_future()),
-                    Some(peer) => Box::new(
-                        server_obj_c_c.borrow_mut().send_ping_req(peer)
-                            .map(|()| future::Loop::Continue(iter))
-                            .or_else(|_| Ok(future::Loop::Break(kbucket_c_c)))
-                    ),
-                }
-            })
-            .map(|_| ());
-
-            handle.spawn(packet_sender);
-            Ok(())
+        // println!("ping wakeup");
+        let kbucket_c = server_obj_c.borrow().kbucket.clone().iter();
+        let server_obj_c_c = server_obj_c.clone();
+
+        let all_sends = kbucket_c.map(|peer|
```

Comment:
good job! Now rename it into `Server::send_pings` that will return a future that sends pings to all it's clients